ddp-workspace/projects/ddp-sdk/src/lib/services/activity/validators/dateValidators/activityDayRequiredDateValidationRule.ts

```diff
@@ -14,10 +14,13 @@ export class ActivityDayRequiredDateValidationRule extends ActivityAbstractValid
             if (!this.isBlank(value) && value.day === null) {
                 this.result = this.message;
                 return false;
+            } else {
+                this.result = null;
+                return true;
             }
         }
-        this.result = null;
-        return true;
+        this.result = this.message;
+        return false;
```

Comment:
OK. This code looks wrong to me.
Most of our answer validations (including this one) are setup so that, if there is no answer available yet for a given field, it just return true (the explanation is that the validation in only applicable once we have a value for it). Here we are undoing this behavior.

@LKapanad Do we have a JIRA ticket for a bug that this code is fixing? This code has been in place for a very long time and since we have had many date fields validated, so it would be weird that this would become a problem all of the sudden.

Reply:
@MocanaAtBroad I think we should close the PR, there was a bug on FON that birthday required validation didn't work correctly because there was no REQUIRED validation on the backend config, and I thought it was a frontend bug. REQUIRED rule was added yesterday and it works now without any changes on frontend.

Reply:
Cool!